src/main/java/com/terraformersmc/terrestria/feature/arch/CanyonArchGenerator.java

```diff
@@ -0,0 +1,169 @@
+package com.terraformersmc.terrestria.feature.arch;
+
+import com.terraformersmc.terraform.noise.OpenSimplexNoise;
+import com.terraformersmc.terrestria.init.TerrestriaFeatures;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.structure.StructureManager;
+import net.minecraft.structure.StructurePiece;
+import net.minecraft.util.math.BlockBox;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.gen.chunk.ChunkGenerator;
+
+import java.util.Random;
+
+public class CanyonArchGenerator extends StructurePiece {
+
+	private OpenSimplexNoise noise;
+
+	private int lineSlope;
+	private int maxHeight;
+	private int radius;
+	private int yStart;
+
+	private int centerX;
+	private int centerZ;
+
+	CanyonArchGenerator(Random random, int centerX, int centerZ, Biome biome) {
+		super(TerrestriaFeatures.CANYON_ARCH_PIECE, 0);
+		this.setOrientation(null);
+
+		this.centerX = centerX;
+		this.centerZ = centerZ;
+
+		int seed = random.nextInt(10000);
+
+		noise = new OpenSimplexNoise(seed);
+
+		lineSlope = seed % 10;
```

Comment:
Hmm, we probably want to generate a random angle instead and then derive x/z factors with sin and cosine. Then these could be used for the values of `a` and `b` in `ax + by + c` for the line test below.

Reply:
makes sense, which one would be faster?

Reply:
I decided against that, instead I'll just generate a and b as random floats